proxy/http/HttpSM.cc

```diff
@@ -3309,7 +3309,7 @@ HttpSM::tunnel_handler_ua(int event, HttpTunnelConsumer *c)
 
     // only external POSTs should be subject to this logic; ruling out internal POSTs here
     bool is_eligible_post_request = (t_state.method == HTTP_WKSIDX_POST);
-    if (is_eligible_post_request) {
+    if (is_eligible_post_request && ua_session) {
```

Comment:
I don't know this code much, but seeing that we don't crash here, presumably ua_session generally is not NULL here. So maybe a better thing is a release assert?